### PR TITLE
Update go-connections dependency

### PR DESCRIPTION
**Pull request by vdemeester**
🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: vendor/github.com/docker/go-connections/tlsconfig/certpool_go17.go [hunk 1]
```diff
@@ -5,16 +5,13 @@ package tlsconfig
...
-
-	"github.com/Sirupsen/logrus"
 )
 
 // SystemCertPool returns a copy of the system cert pool,
 // returns an error if failed to load or empty pool on windows.
 func SystemCertPool() (*x509.CertPool, error) {
 	certpool, err := x509.SystemCertPool()
 	if err != nil && runtime.GOOS == "windows" {
-		logrus.Infof("Unable to use system certificate pool: %v", err)
```
Comment (AkihiroSuda) [inline vendor/github.com/docker/go-connections/tlsconfig/certpool_go17.go]: Can we somehow propagate this error?
e.g. `SystemCertPool() (pool *x509.CertnPool, warn, err error)`

**Review by boaz0 — APPROVED** [review]
👍

File: vendor/github.com/docker/go-connections/tlsconfig/certpool_go17.go (hunk 1 above)
Comment (mlaventure) [inline vendor/github.com/docker/go-connections/tlsconfig/certpool_go17.go]: Would need a PR in docker/go-connections first ;-)

**Review by mlaventure — APPROVED** [review]
LGTM

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**